src/Http/Middleware/FlarumPromotionHeader.php

```diff
@@ -9,25 +9,26 @@
 
 namespace Flarum\Http\Middleware;
 
+use Flarum\Foundation\Config;
 use Psr\Http\Message\ResponseInterface;
 use Psr\Http\Message\ServerRequestInterface;
 use Psr\Http\Server\MiddlewareInterface as Middleware;
 use Psr\Http\Server\RequestHandlerInterface;
 
 class FlarumPromotionHeader implements Middleware
 {
-    public static $enabled = true;
+    protected $enabled = true;
 
-    public static function disable()
+    public function __construct(Config $config)
     {
-        static::$enabled = false;
+        $this->enabled = $config['promote'] ?? true;
```

Comment:
```suggestion
        $this->enabled = $config['flarumPromotionalHeader'] ?? true;
```

Reply:
I feel like that suggested name is a little long? I'm perfectly fine with simple "promote" if we did want to explain it more maybe "promotionHeader"

Reply:
I like it!